src/mono/wasm/runtime/library_mono.js

```diff
@@ -338,6 +339,11 @@ var MonoSupportLib = {
 			return exception_obj ;
 		},
 
+		mono_wasm_add_lazy_load_files: function(lazy_loaded_files) {
```

Comment:
Can this be called while we are paused for some other reason, like a breakpoint? What does `debugger;` do in that case?

Reply:
If it won't send a `debugger.Paused` again, then we won't hit the relevant code path in the proxy. If it does send it, then we wouldn't want to send `Debugger.resume` at the end of that.

Reply:
I tried this out on an E2E scenario by placing a breakpoint in the call to `LazyAssemblyLoader.LoadAssembliesAsync`. This is the C# method that makes the JS interop call to this method. I stepped into the code incrementally and it does dispatch the `Debugger.pause` event. The scripts are parsed and I am able to see them under the source view.

A `Debugger.resume` event does get dispatched but with the order of events it doesn't seem to have a break in the experience (e.g. it gets called but then a `StepInto` command is issued).

So practically, I'm not sure if there is a way for the user to end up in a position where a breakpoint is active while this method is being invoked, although I might be missing something in this trial I ran.

Reply:
Is it possible to insert a breakpoint on cs file that will load the assembly dynamically? And try to step over? In this case would we break the step over of the user with our forced resume?